### DISCUSSION TITLE
Lazy evaluation during debug

### DISCUSSION THREAD

**Discussion by benmatwil** [microsoft/debugpy#1189] [Enhancements]
Would it be possible to have lazy/on demand loading of variables during a debug session similar to the option in pycharm.
For large dataframes, the eager loading of its repr means that stepping through code line by line can be slow when debugging and you receive the warning each time about the slow repr on each line. By only loading the variables on demand, I assume this could alleviate this issue.

**Comment by int19h** [comment]
To clarify, this would be a kind of flag (e.g. in launch.json) that would make it so that no variable shows its value by default, and you have to click or expand something to cause eval? Or are you thinking of something more fine-grained?

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
